articles/node_lifecycle.md

```diff
@@ -27,224 +27,447 @@ Last Modified: {% if page.last_modified %}{{ page.last_modified }}{% else %}{{ p
+For an asynchronous user transition function, the function should additionally receive some method or handler to respond to the transition at a later time (e.g., a handler similar to a `ROS 2 Action GoalHandle`).
+When calling an asynchronous transition callback, the Lifecycle backend relinquishes control of the executor thread to the transition callback and does not expect it back until response.
+The user should be able to send this response from any thread at any point of execution.
+The user will only be able to send a single response with any subsequent responses being deemed invalid (e.g., atomically invalidating the handler).
+Finally, the user will have the ability to atomically check if the method of response is still valid.
 
-### Transition State: Activating
+#### Cancelling a Transition
 
-In this transition state the callback `onActivate` will be executed.
-This method is expected to do any final preparations to start executing.
+A cancel service or equivalent will be available to allow cancelling mid-transition.
+This is the only built-in interface for an external node to interact with a node in a Transition State.
+
+The cancellation service will follow a [cooperative cancellation approach](https://www.drdobbs.com/parallel/interrupt-politely/207100682). Similar to a transition, a cancel request is serviced on a first-come, first-serve basis with respect to time received. To avoid accidental cancellations, the cancel request must include the Transition State `id` (or equivalent identifier) of the desired transition to cancel.
+This follows the same principles of [concurrent `PUT` requests for REST APIs](https://medium.com/swlh/handling-concurrent-requests-in-a-restful-api-5a25a4b81a1). A cancel request is considered valid if the ongoing transition matches this `id` as well as there is not already an ongoing cancel request (similar to the [Definition of a Transition](#definition-of-a-transition)). 
+
+Once a cancel is valid and active, it is up to the user code to check on this cancel request (similar to a `ROS 2 Action`).
+This is designed as only the user can demarcate appropriate times and code paths to clean up an ongoing transition (i.e., only the user knows how to safely clean up a request at any given part of the ongoing transition).
+The user can opt to ignore this request (i.e., the transition continues as if no cancel request existed) or attempt to handle the request.
+The user can denote a successful handling of the request or a failed handling.
+
+If the user accepts and successfully handles the cancellation request, the state will follow the `CallbackReturn::FAILURE` path of the state machine. 
+This follows the "fallback" behavior of a transition.
+If the user accepts and unsuccesfully handles the cancellation request, the state will follow the `CallbackReturn::ERROR` path of the state machine. 
+An ignored cancellation request becomes inactive and failed the instant a transition is inactive.
```

Comment:
```suggestion
An ignored cancellation request becomes inactive and failed the instant a transition is inactive.
```

I would think that we'd basically transition back down to what the cancel would have done for us if it were processed properly. E.g. If transitioning to Active and cancel is ignored, we just force the state machine back down to Configured. 

Or, if cancel accepted + processed puts it always back in the unconfigured state, then an ignored cancel request would transition the system back down the unconfigured automatically so it seems as if it were handled regardless